nilearn/decoding/decoder.py

```diff
@@ -677,29 +667,21 @@ def fit(self, X, y, groups=None):
             cv_object = LeaveOneGroupOut()
 
         else:
-            cv_object = check_cv(cv, y=y, classifier=self._is_classification)
+            cv_object = check_cv(cv, y=y, classifier=is_classifier(self))
```

Comment:
Starting from line 654 until 672, I thought an abstract method like `_setup_cv` could be used, and implemented in concrete classes.